pkg/compare/container.go

```diff
@@ -0,0 +1,140 @@
+package compare
+
+import (
+	"fmt"
+	"os/exec"
+	"path/filepath"
+	"strings"
+)
+
+type engine struct {
+	name         string
+	requiresSudo bool
+	containerID  string
+	tempDir      string
+}
+
+// isContainer reports whether the given path is a reference to a file in a container by verifying if it starts with "container://".
+func isContainer(path string) bool {
+	return strings.HasPrefix(path, "container://")
+}
+
+// parsePath returns the image and referencePath (path to the directory for metadata.yaml), given a path
+// of the form container://<IMAGE>:<TAG>:/path_to_metadata.yaml
+func parsePath(path string) (string, string, error) {
```

Comment:
Returning a tuple can sometimes be confusing, because you have to remember which string is which.  Sometimes it's valuable to create a struct with named arguments instead:

```suggestion
type parsedPath struct {
    image string
    path  string
}

func parsePath(path string) (parsedPath, error) {
```